add note on views / copies in getting started -- #2224 (#2226)
JuliaData/DataFrames.jl: docs/src/man/getting_started.md

a function object that tests whether each value belongs to the subset
(partial application of `in`): `df[in([1, 5, 601]).(df.A), :]`.

!!! note

    As with matrices, subsetting from a data frame will usually return a copy of
    columns, not a view or direct reference.

>     The only subsetting situations where data frames will **not** return a copy are:

Reviewer: ```suggestion
    The only indexing situations where data frames will **not** return a copy are:
```

>     - when a `!` is placed in the first indexing position (`df[!,:A]`, or `df[!, [:A, :B]]`),

Reviewer: ```suggestion
    - when a `!` is placed in the first indexing position (`df[!, :A]`, or `df[!, [:A, :B]]`),
```

    - when using `.` (`getpropery`) notation (`df.A`),

>     - when a single row is selected using an integer (`df[1, :A]`), or

Reviewer: ```suggestion
    - when a single row is selected using an integer (`df[1, [:A, :B]]`)
```

    - when `view` or `@view` is used (e.g. `@view df[1:3, :A]`).

>     More details on copies, views, and references can be found [here.](https://juliadata.github.io/DataFrames.jl/stable/lib/indexing/#getindex-and-view-1)

Reviewer: ```suggestion
    More details on copies, views, and references can be found
    [here.](https://juliadata.github.io/DataFrames.jl/stable/lib/indexing/#getindex-and-view-1)
```

a function object that tests whether each value belongs to the subset
(partial application of `in`): `df[in([1, 5, 601]).(df.A), :]`.

!!! note

>     While data frames support indexing syntax that is very similar to

Reviewer: The text seems to say that this is different from matrices. Actually this is exactly the same - normally indexing into a data frame creates a copy like for matrices.

Author: OH! Sorry, moving back and forth between numpy and julia too much these days...

>     While `DataFrames` support indexing syntax that is very similar to

Reviewer: ```suggestion
    While data frames support indexing syntax that is very similar to
```

>     matrices, subsetting from a `DataFrame` will return a copy, not

Reviewer: I would say a copy of what - i.e. of columns.

>     matrices, subsetting from a `DataFrame` will return a copy, not

Reviewer: it is not clear to me what you mean by "subsetting" (I am not a native though)

Author: "Subsetting" would be any operation that returns a subset of the original data frame. 

This note is in the section called "taking a subset", so I feel like the term is appropriate in context? I can change it to "when you take a subset of a data frame" if you'd like.

    a view.

    `DataFrames` will only return a view when:

>     - a `!` is placed in the first indexing position (`df[!,:A]`, or `df[!, [:A, :B]]`),

Reviewer: the first two options are not views - it is either a direct access to a vector stored in the data frame or creation of a new data frame without copying of the data.

Author: RIGHT! OK, then how about "`DataFrames` will not create a copy when:"

Reviewer: this is OK, but DataFrames.jl not `DataFrames` (as we would talk about the functionality of the package).

Please always remember that `DataFrame` is only one concrete type we provide. We also provide `SubDataFrame` which is also a data frame.

Author: > Please always remember that DataFrame is only one concrete type we provide. We also provide SubDataFrame which is also a data frame.

right!

>     - when using `.` notation (`df.A`),

Reviewer: maybe better write `getproperty`?

Author: I don't think most people know what `getproperty` refers to (I don't!). I can put `getproperty` in parentheses?

Reviewer: That is why I have written to wait for someone else to comment. I can make sure that the text is correct with respect to the implementation and it would be good to have someone else to comment on the wording :smile:.

>     - when a single row is selected using a single integer (`df[1, :A]`), or

Reviewer: ```suggestion
    - when a single row is selected using an integer (`df[1, :A]`), or
```

Author: I feel like there's value in `a single` to emphasize that this doesn't apply for collections of integers (e.g. `df[[1,2], :A]`). You are correct that `an integer` implies just one, but not very strongly.

Reviewer: ok - let us wait what to hear native speakers feel :smile:. Thank you for caring for such details.

>     - when the `@view` macro is used (`@view df[1:3, :A]`).

Reviewer: ```suggestion
    - when `view` or `@view` is used (e.g. `@view df[1:3, :A]`).
```

>     More details can be found in [`getindex` and `view`](@ref).  

Reviewer: This link will not be correct. I would rather direct a reader to the section in the manual I have shared with you.

Author: Oh, ok -- sorry, I thought that was a link to that section (when I build locally links don't seem to work, so kinda in the dark...).

Reviewer: yes - documenter is tricky. I think that the easiest thing is just to copy paste the physical link to master on GitHub (this is not ideal, but I assume we are not going to change this functionality in the near future so it should be good enough to do it this way).

Author: OK, thanks.
